src/glue/jobs/s3_to_json.py

```diff
@@ -522,74 +526,91 @@ def write_file_to_json_dataset(
     Returns:
         list: A list of files uploaded to S3
     """
-    # Configuration related to where we write our part files
-    part_dir = os.path.join(
+    # Configuration related to where we write the JSON
+    output_dir = os.path.join(
         f"dataset={metadata['type']}", f"cohort={metadata['cohort']}"
     )
-    os.makedirs(part_dir, exist_ok=True)
+    os.makedirs(output_dir, exist_ok=True)
     part_number = 0
-    output_path = get_part_path(
-        metadata=metadata, part_number=part_number, part_dir=part_dir, touch=True
+    # we update this value as we write
+    current_part_path = get_output_path(
+        metadata=metadata, part_number=part_number, output_dir=output_dir, touch=True
+    )
+    compressed_output_path = get_output_path(
+        metadata=metadata, part_number=None, output_dir=output_dir, touch=True
     )
 
     # We will attach file metadata to the uploaded S3 object
     s3_metadata = _derive_str_metadata(metadata=metadata)
+    line_count = 0
     uploaded_files = []
-    with z.open(json_path, "r") as input_json:
-        current_output_path = output_path
-        line_count = 0
+    # fmt: off
+    # <python3.10 requires this backslash syntax, we currently run 3.7
```

Comment:
Thanks, we do have a ticket to update to python 3.10, do you think we should do that sooner than later?

Reply:
No, we may end up abandoning Glue altogether so it doesn't make sense spending time upgrading python just to avoid these minor inconveniences.